src/mango/depository.ts

```diff
@@ -290,4 +291,103 @@ export class MangoDepository {
+    mango: Mango
+  ): Promise<{ yield: number, feeAmount: number, slippageAmount: number, priceImpact: number }> {
+    const perpPrice = await this.getCollateralPerpPriceUI(mango);
+    const takerFee = this.getCollateralPerpTakerFees(mango);
+    // This is not the actual price, just the highest price it has to go to fill, so execution will be slightly better
+    const priceImpact = await this.getRedeemingPriceImpact(redeemableQuantity, mango);
+    if (!priceImpact) {
+      throw "Error: cannot fill";
+    }
+    const collateralQuantity = redeemableQuantity / perpPrice;
+    const perfectAmount = collateralQuantity * perpPrice;
+    const realAmount = collateralQuantity * priceImpact;
+    const feeAmount = realAmount * takerFee;
+    const slippageAmount = (perfectAmount - realAmount) - feeAmount;
+    const finalYield = realAmount - feeAmount;
+    return { yield: finalYield, feeAmount: feeAmount, slippageAmount: slippageAmount, priceImpact: priceImpact };
+  }
+
+  // Price impact for a minting operation (Mint or RebalancingLite when PnlPolarity is positive)
+  public async getMintingPriceImpact(
+    collateralQuantity: number,  // UI units
+    mango: Mango
+  ): Promise<number | undefined> {
+    // Minting is placing a ask IoC limit order, will check the impact on the bid side
+    return this.getMangoPerpOrderBookPriceImpact(mango, collateralQuantity, "bid");
+  }
+
+  // Price impact for a redeeming operation (Redeeming or RebalancingLite when PnlPolarity is negative)
+  public async getRedeemingPriceImpact(
+    redeemableQuantity: number, // UI units
+    mango: Mango
+  ): Promise<number | undefined> {
+    const takerFee = this.getCollateralPerpTakerFees(mango);
+    // Remove the max possible fees, as we place an order on that amount to  make sure we can pay for the fees on the program side.
+    const redeemableQuantityMinusFees = redeemableQuantity * (1 - takerFee);
+    // Use latest perp price
+    const collateralPerpPrice = await this.getCollateralPerpPriceUI(mango);
+    const collateralQuantity = redeemableQuantityMinusFees / collateralPerpPrice;
+
+    if (!collateralQuantity) {
+      console.log("Error: cannot derive collateral Quantity");
+      return undefined;
+    }
+    // Redeeming is placing a bid IoC limit order, will check the impact on the ask side
+    return this.getMangoPerpOrderBookPriceImpact(mango, collateralQuantity, "bid");
+  }
+
+  // Return the price impact for an order on a 
+  public async getMangoPerpOrderBookPriceImpact(
+    mango: Mango,
+    quantity: number, // UI AMount
+    side: "bid" | "ask"
+  ): Promise<number | undefined> {
+    const pmc = this.getPerpMarketConfig(mango); // perpMarketConfig
+    const pm: PerpMarket = await mango.client.getPerpMarket(
+      pmc.publicKey,
+      pmc.baseDecimals,
+      pmc.quoteDecimals
+    ); // perpMarket
+    const nativeQuantity = (new BN(quantity)).mul(new BN(this.collateralMintDecimals).pow(new BN(10)));
+    switch (side) {
```

Comment:
You could rewrite it differently so you don't need to `switch` & avoid the eslint error:

```typescript
const loadSideBook = side === 'bid' ? pm.loadBids : pm.loadAsks;
const sideBook = await loadSideBook(mango.client.connection, false);
return sideBook.getImpactPriceUi(nativeQuantity);
```

Reply:
Neat